### PR TITLE
Fix: handle profile emojis

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-2646
Fixes RNBW-1255
Fixes RNBW-3778

## What changed (plus any additional context for devs)

In addition to the bug described on linear,

- When creating new accounts, we are not sending an `accountSymbol` param to the backend, this case is not handled correctly by backend so the profile is not created, then every time the emoji or color is being updated in the app, we try to update it on the backend as well but is rejected from since there was nothing to update in the first place. I was able to repro this in the testflight app
- when creating or watching new addresses we aren't looking to the backend for symbol and colors, making the original linear ticket to happen, now we check for backend first 

Additionally I changed the default avatar color to gray, which is going to appear while waiting for the web profile, if any. Let me know what you think about it 

## PoW (screenshots / screen recordings)

https://www.loom.com/share/f4fd793eca4d41dd85967e15897718a9

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Comment by osdnk** [comment]
I don't really understand those changes and I feel sad I have to review this. Maybe we can have a quick huddle about this so I can approve then?

**Comment by terrysahaidak** [comment]
I was looking at the error handling here and noticed that `getPreference` actually logs in sentry `error setting preference` - seems to be copy-paste from `setPreference`

**Comment by osdnk** [comment]
I know we're all busy, but kindly pinging about this

**Comment by estebanmino** [comment]
yeah sorry my bad, leaving a video with a general explanation of what this PR is of https://www.loom.com/share/419ca24b3f174e2685b3de9cd7637c1e

**Comment by estebanmino** [comment]
@osdnk  yeah sorry my bad, leaving a video with a general explanation of what this PR is of https://www.loom.com/share/419ca24b3f174e2685b3de9cd7637c1e

File: src/components/expanded-state/WalletProfileState.js [hunk 1]
```diff
@@ -147,6 +172,14 @@ export default function WalletProfileState({
     inputRef,
   ]);
 
+  useEffect(() => {
+    const getProfile = async () => {
+      const profile = await getWebProfile(address);
+      setWebProfile(profile || {});
```
Comment (osdnk) [inline src/components/expanded-state/WalletProfileState.js:178]:
```suggestion
      setWebProfile(profile ?? {});
```

File: src/components/profile/AvatarCircle.js [hunk 2]
```diff
@@ -36,20 +36,22 @@ export default function AvatarCircle({
...
   } = useAccountProfile();
+
   const accountSymbol = showcaseAccountSymbol || profileAccountSymbol;
   const resolvedColor =
     showcaseAccountColor != null
       ? typeof showcaseAccountColor === 'string'
         ? showcaseAccountColor
         : colors.avatarBackgrounds[showcaseAccountColor]
-      : colors.avatarBackgrounds[profileAccountColor || 0];
+      : colors.avatarBackgrounds[(!newProfile && profileAccountColor) || 10];
```
Comment (osdnk) [inline src/components/profile/AvatarCircle.js:54]:
```suggestion
      : colors.avatarBackgrounds[(!newProfile && profileAccountColor) ?? 10];
```

**Review by osdnk — APPROVED** [review]
after cleaning-up looks good go to me

File: src/components/expanded-state/WalletProfileState.js [hunk 3]
```diff
@@ -81,27 +88,45 @@ export default function WalletProfileState({
...
+    const indexOfForceColor = colors.avatarBackgrounds.indexOf(forceColor);
+    return forceColor
+      ? forceColor
+      : isNewProfile && address
+      ? profileUtils.addressHashedColorIndex(address)
+      : profile.color !== null
+      ? profile.color
+      : isNewProfile
+      ? null
+      : (indexOfForceColor !== -1 && indexOfForceColor) || getRandomColor();
```
[the remark this answers is not in the document]
Comment (estebanmino) [reply src/components/expanded-state/WalletProfileState.js:120]: hm yeah i didn't want to touch it

**Review by skylarbarrera — APPROVED** [review]
🙆‍♂️

**Review by terrysahaidak — APPROVED** [review]

**Comment by estebanmino** [comment]
video for @ibrahimtaveras00  https://www.loom.com/share/118e6fa28b30462d9979ff220bd048a1

**Comment by estebanmino** [comment]
`qa-later` approved by @ibrahimtaveras00, gonna merge

**Pull request merged**